source/indices.d.ts

```diff
@@ -0,0 +1,21 @@
+/**
+Create a tuple type of an array's indices.
+
```

Comment:
There needs to be a text explanation with some use-cases where this type could be useful.